include/deal.II/base/vectorization.h

```diff
@@ -68,6 +68,27 @@ namespace std
 
 DEAL_II_NAMESPACE_OPEN
 
+namespace internal
+{
+  template <typename T>
+  struct NumberType 
```

Comment:
@MathiasMen you don't want to move all three there, only the first two, leaving the one for `VectorizedArray` here. There are some circular include dependencies if you try to move everything, that's @jppelteret 's experience, if I am correct.